pallets/runtime/tests/src/transaction_payment_test.rs

```diff
@@ -142,17 +149,24 @@ fn signed_extension_transaction_payment_is_bounded() {
         .execute_with(|| {
             let user = AccountKeyring::Bob.to_account_id();
             let free_user = Balances::free_balance(&user);
+
+            // Get the current weight settings.
+            let weights = <TestStorage as frame_system::Config>::BlockWeights::get();
+            let per_byte =
+                <TestStorage as pallet_transaction_payment::Config>::TransactionByteFee::get();
+
+            // Calculate maximum transaction fee.
+            let base_fee = weight_to_fee(weights.get(DispatchClass::Normal).base_extrinsic);
+            let len_fee = per_byte.saturating_mul(10);
+            let max_block_fee = weight_to_fee(weights.max_block);
+            let max_fee = base_fee + len_fee + max_block_fee;
```

Comment:
Could you move this into a standalone function? Seems like a reusable component.

Reply:
There is only one use-case for this.  No need to break it  out into a function.